src/models/CommonModel.ts

```diff
@@ -123,6 +124,31 @@ export class CommonModel extends CommonSchema<CommonModel> {
     }
   }
 
+  /**
+   * Adds a tuple to the model.
+   * 
+   * If a item already exist it will be merged.
+   * 
+   * @param tupleModel 
+   * @param schema 
+   * @param index 
+   */
+  addItemTuple(tupleModel: CommonModel, schema: Schema, index: number): void {
+    let modelItems = this.items;
+    if (!Array.isArray(modelItems)) {
+      Logger.warn('Trying to add item tuple to a non-tuple item, will drop existing item model', tupleModel, schema, index);
```

Comment:
Maybe there are cases, when we should throw an error, when items were single schema and then we change to tuple 🤔

Reply:
Because of the restrictiveness of tuples, that should always be preferred over simple array type. So I feel like this is the correct approach.

In which cases would you rather have the generation fail? 🤔

Reply:
This logic for tuple/items will be only when user will define combined schema like `oneOf/anyOf` etc, then we should consider it as union type, not override it, but at the moment, as I remember, we don't support union types explicit :) Am I right? 😅

Reply:
You are, changed it.